New package: hunspell-el_GR-0.4.0

protesilaos opened the pull request:
My very first package. Please let me know if something is amiss and I will try to fix it.

flexibeast commented:
Thanks for your contribution!

Unfortunately, the build system has reported [xlint issues](https://travis-ci.org/void-linux/void-packages/builds/610896763?utm_source=github_status&utm_medium=notification). Please be sure to run `xlint`, from the `xtools` package, as described in [CONTRIBUTING](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md), before submitting a new package template or template update. (You should also make sure to read all of CONTRIBUTING.)

More generally, have you tried building your package with `./xbps-src pkg hunspell-el_GR`?  Your `distfiles=` line doesn't specify which file to download, so i would expect the build to fail.

There was actually [a PR for `hunspell-el_GR` several days ago](https://github.com/void-linux/void-packages/pull/16067), which the submitter has since closed for some reason. However, @Johnnynator [noted](https://github.com/void-linux/void-packages/pull/16067#discussion_r341855317) that the license should be 'GPL-2.0-only".

protesilaos commented:
Thank you for the valuable feedback. Since there already is a PR for this, I think it would be better to withdraw mine, right? Or perhaps just fix the errors and proceed?

I am okay either way. Assuming the latter…

`./xbps-src pkg hunspell-el_GR` builds the package as expected.  Then I can install it normally with `xi`. I am actually using the software already.

I did run `xlint srcpkgs/hunspell-el_GR/template` before making the PR and it offered some feedback with regard to the order of the various fields. Once I addressed all the warnings it stopped returning any output. I assume this to be a good thing.

I just ran the commands again and everything works normally.

I see that Travis lists an error about the license. I do not get this locally though, but it should concern the fact that I do not install a license file. The reason is that the licensing of this package is rather obscure and convoluted. The `LICENSE` it offers suggests that the package _itself_ is distributed under the 3-clause BSD, but inside of it the files have their own licenses. The dictionaries I declare are under the GPL, though no version is specified. While the Libreoffice page for this software writes that the license is, in fact, the LGPL-2.1.

I see that this is also mentioned in the links you provide. I am fine with `GPL2-only`.

Please let me know how we should proceed from here.

Chocimier commented on srcpkgs/hunspell-el_GR/template:
```diff
@@ -0,0 +1,24 @@
+distfiles="${homepage}/files/latest/download"
+checksum=541328be3043d861361fd8d79874ac8d4cd042b657bbd9ba94a5dafab9cbae2d
+
+post_extract() {
```
extracting was organized better in that PR.

Chocimier replied:
still

protesilaos replied:
Okay, I will follow that method.

Chocimier commented on srcpkgs/hunspell-el_GR/template:
```diff
@@ -0,0 +1,24 @@
+maintainer="Protesilaos Stavrou <info@protesilaos.com>"
+license="GPL-3"
+homepage="https://sourceforge.net/projects/orthos-spell"
+distfiles="${homepage}/files/latest/download"
```
Distfiles should point to exact version.

protesilaos commented:
Thank you for the feedback! I addressed all the issues. The package built, installed, linted properly. I can see that Travis accepts it now. Please let me know if you need me to do anything else about it.

Chocimier commented on srcpkgs/hunspell-el_GR/template:
```diff
@@ -0,0 +1,24 @@
+pkgname=hunspell-el_GR
+version=0.4.0
+revision=1
+_version=0.4.0-87
```
I would rather do something similar to libcli, so only `version` would be needed.

protesilaos replied:
Okay, I will amend it accordingly.

protesilaos commented:
Thank you @Chocimier. I reviewed the template following your guidance. Now it is essentially the same as the previous closed PR. As such, I did not have to implement your suggestion about the parameter expansion used in `libcli` version.

I did the commit fixup as well. Hopefully it looks right and everything is just merged into a single commit.
